src/Cell.cpp

```diff
@@ -583,8 +619,9 @@ CellBasic* CellBasic::clone() {
 
 /**
  * @brief Subdivides the Cell into clones for fuel pin angular sectors.
+ * @param subcells an empty vector to store all subcells
  */
-void CellBasic::sectorize() {
+void Cell::sectorize(std::vector<Cell*>* subcells) {
```

Comment:
Well, passing the vector itself would do an *implicit copy* of the vector which we do not want here. In other words, the vector that is used by the ``sectorize`` routine will be a deep copy of the one that was passed into it as a parameter. The ``sectorize`` routine would then modify / extend the vector copy with ``Cell`` rings, but the caller's vector would be left untouched. There are two ways to ensure that the caller receives the updated vector - 1) pass in a vector pointer as done here, or 2) return the updated vector copy from ``sectorize``. The first option requires less overhead (memory and time to make the vector copy) so I went with it here. But I agree with you that in most places in the code it has not been necessary to use vector / map pointers, but that is simply because our vector / map variables are typically private and only used by a particular class instance (rather than being passed around between instances).